src/pages/allPosts/index.tsx

```diff
@@ -0,0 +1,45 @@
+import { useEffect, useState } from 'react';
+import Layout from '../../components/Layout';
+import UserPostCard from '../../components/UserPostCard';
+import { getAllPosts } from '../../fetch/fetchPosts';
+import { getPostImage } from '../../util/getImages';
+import './allposts.css';
+const AllPosts = () => {
+  const mockId = 24;
+  const [isLoading, setIsLoading] = useState(true);
+  const [postsInformationArray, setPostsInformationArray] = useState([]);
+  const [imageInformation, setImageInformation] = useState({
+    download_url: '',
+  });
+
+  useEffect(() => {
+    getAllPosts().then((postData) => {
+      setPostsInformationArray(postData);
+    });
+    getPostImage(mockId).then((imageInfo) => {
+      setImageInformation(imageInfo);
+    });
+    setIsLoading(false);

```

Comment:
We probably only want to stop loading after we've fetched and set the posts array right? This code here says that loading will be set to false as soon as we make the request rather than after it's returned. Also we probably want to fetch an image for each post based on the post's id, this code here just gets 1 image for all posts. We should fix that

Reply:
Yes i needed to change the setIsLoading to false inside the getAllPosts promise and use a different logic to match an image id with post id because i'm trying to use different Api endpoints and here my logic failed, thank you i will improve this next time.